pkg/multicluster/clustercache/cache_manager.go

```diff
@@ -0,0 +1,89 @@
+package clustercache
+
+import (
+	"context"
+	"sync"
+
+	"github.com/solo-io/solo-kit/pkg/errors"
+	"github.com/solo-io/solo-kit/pkg/multicluster/handler"
+	"k8s.io/client-go/rest"
+)
+
+//go:generate mockgen -destination=./mocks/cache_manager.go -source cache_manager.go -package mocks
+
+type cacheWrapper struct {
+	cancel    context.CancelFunc
+	coreCache PerClusterCache
+}
+
+type PerClusterCache interface {
+	IsPerCluster()
```

Comment:
can you add some comments here to describe the intent / effect of this interface?

Reply:
nit: consider renaming `PerClusterCache` to `ClusterCache` and `IsPerCluster()` to `IsClusterCache()`